### PR TITLE
Enforce Mandatory Client ID (Github Issue #704)

### PR DESCRIPTION
**Pull request by sming**
### Use Case Resolved: 

- I am a Heroic developer
- who wants to know who is querying the Heroic API
- so that when Heroic's performance suffers due to egregious query patterns, we know who to contact

### Design & Implementation Notes

* See [[RFC] Enforce Mandatory Client ID](https://docs.google.com/document/d/13aZjEpRqeD9YfjRknWMTz_fzdGd3E7S8XJNI20_Rxcc/edit#heading=h.f4rxemuawl3) for detailed coverage of the Use Case outlined above.
* but in a nutshell, we opted for Envoy but it didn't have the features necessary.
* hence we reverted to using Java request filters in the Heroic API codebase.

### The Deployment Plan

From [Plan of Action](https://docs.google.com/document/d/13aZjEpRqeD9YfjRknWMTz_fzdGd3E7S8XJNI20_Rxcc/edit#heading=h.alzqf34ymh6e) : 

1. ~~Send a warning email to all of engineering that says uses have 1 month to start setting Client ID to something accountable~~
1. ~~Simultaneously, send the same message to #monitoring-users~~
1. ~~after 1 month, send the messages again~~
1. push to canaries and try out some non-compliant and compliant requests to check it’s working
1. Mop up any lingering questions/problems on #monitoring-users
1. deploy to all heroic instances
1. field any issues that arise

### CODE REVIEW

**Comment by sming** [comment]
Reopening because we're "pivoting" (yet again ¯\_(ツ)_/¯) back to doing this filtering in Heroic.

File: metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableMetricModule.java [hunk 1]
```diff
@@ -179,9 +179,9 @@ public int getMaxWriteBatchSize() {
                 }
 
                 @Override
-                public AsyncFuture<Void> destruct(final BigtableConnection value) {
+                public AsyncFuture<Void> destruct(final BigtableConnection t) {
```
Comment (sming) [inline metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableMetricModule.java:182]: fixed compiler warning

File: metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableMetricModule.java [hunk 2]
```diff
@@ -141,11 +141,11 @@ public BigtableMetricModule(
     }
 
     @Override
-    public Exposed module(PrimaryComponent primary, Depends depends, String id) {
+    public Exposed module(PrimaryComponent primary, Depends backend, String id) {
```
Comment (sming) [inline metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/BigtableMetricModule.java:144]: fixed compiler warning

**Review by malish8632 — APPROVED** [review]
+1

**Pull request merged**